heimlig/src/common/jobs.rs

```diff
@@ -1,21 +1,33 @@
 use crate::hsm::keystore;
 use crate::hsm::keystore::Id;
 
-#[derive(Clone, Eq, PartialEq, Debug)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
 pub enum Error {
     /// The amount of requested data was too large.
     RequestTooLarge,
     /// A cryptographic error occurred.
     Crypto(crate::crypto::Error),
+    /// No key store present.
+    NoKeyStore,
     /// A key store error occurred.
     KeyStore(keystore::Error),
 }
 
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum RequestType {
+    ImportKey,
+    GetRandom,
+    EncryptChaChaPoly,
+    EncryptChaChaPolyExternalKey,
+    DecryptChaChaPoly,
+    DecryptChaChaPolyExternalKey,
+}
+
```

Comment:
I have found this issue before, i.e., having to declare a separate "simple" enum to represent the possible variants of a proper sum type enum. I wonder if there's a crate that implements a macro for this?